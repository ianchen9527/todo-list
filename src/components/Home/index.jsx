import React from "react";
import styles from "./index.module.scss";
import { Link } from "react-router-dom";

const Home = () => {
  return (
    <div className={styles["wrapper"]}>
      <div className={styles["title"]}>
        Welcome to the <b>Proton Note Demo</b>
      </div>
      <Link to="/notes" className={styles["enter-button"]}>
        <div className={styles["logo"]} />
        <div className={styles["description"]}>Enter</div>
      </Link>
    </div>
  );
};

export default Home;
