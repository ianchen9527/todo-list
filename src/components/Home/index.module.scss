@import "../../stylesheets/constants/colors.scss";

.wrapper {
  width: 100vw;
  height: 100vh;
  background-color: $main-color;
  overflow: hidden;
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  align-items: center;

  .title {
    font-size: 40px;
    line-height: 54px;
    margin-bottom: 65px;
    max-width: 90vw;
    text-align: center;
  }

  .enter-button {
    width: 150px;
    height: 200px;
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    align-items: center;
    border: 5px solid $black;
    border-radius: 10px;
    cursor: pointer;
    text-decoration: none;

    .logo {
      width: 120px;
      height: 120px;
      background-image: url("../../assets/images/list.svg");
    }

    .description {
      color: $black;
      font-size: 20px;
      font-weight: bold;
      max-width: 90vw;
      text-align: center;
      text-decoration: none;
    }

    &:hover {
      .logo {
        background-image: url("../../assets/images/list-hover.svg");
      }
    }

    &:visited {
      color: $black;
    }
  }
}
