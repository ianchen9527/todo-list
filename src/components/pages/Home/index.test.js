import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Home from './index';

describe('Home', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('should render without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
