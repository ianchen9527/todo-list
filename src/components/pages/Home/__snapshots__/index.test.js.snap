// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render without crashing 1`] = `
<div
  className="wrapper"
>
  <div
    className="title"
  >
    Welcome to the 
    <b>
      Proton Note Demo
    </b>
  </div>
  <Link
    className="enter-button"
    to={[Function]}
  >
    <div
      className="logo"
    />
    <div
      className="description"
    >
      Enter
    </div>
  </Link>
</div>
`;
